### PR TITLE
MAT-6547: initial pass at HQMF generator code from the MAT

### PR DESCRIPTION
**Pull request by nmorasb**
## MADiE PR

Jira Ticket: [MAT-6547](https://jira.cms.gov/browse/MAT-6547)
(Optional) Related Tickets:

### Summary
- Bring in HQMF Generators from MAT
- Generate HQMF

### All Submissions
* [x] This PR has the JIRA linked.
* [ ] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/madie/hqmf/qdm_5_6/HQMFGenerator.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+@AllArgsConstructor
+public class HQMFGenerator implements Generator {
+
+  private HQMFMeasureDetailsGenerator hqmfMeasureDetailsGenerator;
+  private HQMFDataCriteriaGenerator hqmfDataCriteriaGenerator;
+
+  private final Logger logger = LoggerFactory.getLogger(HQMFDataCriteriaGenerator.class);
+
+  /**
+   * Generate hqmf for CQL Based measures (QDM version 5.5)
```
Comment (adongare) [inline src/main/java/gov/cms/madie/hqmf/qdm_5_6/HQMFGenerator.java]:
```suggestion
   * Generate hqmf for CQL Based measures (QDM version 5.6)
```

**Review by adongare — DISMISSED** [review]

File: src/main/java/gov/cms/madie/hqmf/HQMFAttributeGenerator.java
Comment (jkotanchik-SB) [inline src/main/java/gov/cms/madie/hqmf/HQMFAttributeGenerator.java:1]: There's a number of generation classes that were only for old version of QDM and were left in the MAT for backwards compatibility with old measures.

We should not need these classes in madie.

File: src/main/java/gov/cms/madie/hqmf/HQMFDataCriteriaElementGenerator.java
Comment (jkotanchik-SB) [inline src/main/java/gov/cms/madie/hqmf/HQMFDataCriteriaElementGenerator.java:1]: Hopefully, this class can also be removed.

File: src/main/java/gov/cms/madie/hqmf/HQMFFinalCleanUp.java [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+  private static final String reverseEntryCheckFile = "xsl/final_hqmf_entry_deletion_check.xsl";
+
+  /** The Constant deleteUnUsedEntryFile. */
+  private static final String deleteUnUsedEntryFile = "xsl/hqmf_delete_Unused_entry.xsl";
+
+  /**
+   * Clean.
+   *
+   * @param me the me
+   */
```
Comment (jkotanchik-SB) [inline src/main/java/gov/cms/madie/hqmf/HQMFFinalCleanUp.java]: Repetitive javadocs is how I know it came from the MAT codebase.

File: src/main/java/gov/cms/madie/hqmf/HQMFGeneratorFactory.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+@Slf4j
+@Component
+@AllArgsConstructor
+public class HQMFGeneratorFactory {
+
+  private HQMFGenerator qdm_5_6_generator;
+
+  public Generator getHQMFGenerator() {
+    // Keeping generator around in case we need to support additional versions or implementations in
+    // the future
```
Comment (jkotanchik-SB) [inline src/main/java/gov/cms/madie/hqmf/HQMFGeneratorFactory.java:18]: Comment is not relevant in this context. It mattered in the MAT where we needed to be able generate HQMF files for older versions of QDM.

File: src/main/java/gov/cms/madie/hqmf/HQMFAttributeGenerator.java
Comment (adongare) [reply src/main/java/gov/cms/madie/hqmf/HQMFAttributeGenerator.java:1]: @jkotanchik-SB I think it is deprecated but still being used in src/main/java/gov/cms/madie/hqmf/qdm_5_6/HQMFClauseLogicGenerator.java line 2365 to generate the data criteria for attributes. I'm not sure if this is valid in QDM5.6. if it is not valid then we can remove these two classes

File: src/main/java/gov/cms/madie/hqmf/HQMFGeneratorFactory.java (hunk 3 above)
Comment (adongare) [reply src/main/java/gov/cms/madie/hqmf/HQMFGeneratorFactory.java:18]: hope we don't get another version of QDM(5.7)

File: src/main/java/gov/cms/madie/hqmf/HQMFFinalCleanUp.java (hunk 2 above)
Comment (adongare) [reply src/main/java/gov/cms/madie/hqmf/HQMFFinalCleanUp.java]: me the me allover!

File: src/main/java/gov/cms/madie/hqmf/HQMFGeneratorFactory.java (hunk 3 above)
Comment (jkotanchik-SB) [reply src/main/java/gov/cms/madie/hqmf/HQMFGeneratorFactory.java:18]: Honestly, if we do, it might better to replace this flow with something more manageable.

**Review by jkotanchik-SB — APPROVED** [review]
